### ISSUE TITLE
module fails to build for Linux 6.12: error: macro "_spin_lock_init" requires 2 arguments, but only 1 given

### ISSUE THREAD

**Issue by linuxmaniac** [sipwise/rtpengine#1882]
### rtpengine version the issue has been seen with

12.5.1.15

### Used distribution and its version

Debian Sid

### Linux kernel version used

6.12.3-rt-amd64

### CPU architecture issue was seen on (see `uname -m`)

x86_64

### Steps to reproduce the problem

from https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1089503:

```
Package: rtpengine-kernel-dkms
Version: 12.5.1.15-1
Severity: serious
Tags: upstream

DKMS make.log for rtpengine/12.5.1.15 for kernel 6.12.3-rt-amd64 (x86_64)
Sat Dec  7 23:22:21 UTC 2024
make: Entering directory '/usr/src/linux-headers-6.12.3-rt-amd64'
  CC [M]  /var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.o
/var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.c:181:50: error: macro "_spin_lock_init" requires 2 arguments, but only 1 given
  181 | static inline void _spin_lock_init(_spinlock_t *s) {
      |                                                  ^
In file included from /usr/src/linux-headers-6.12.3-common-rt/include/linux/spinlock.h:455,
                 from /usr/src/linux-headers-6.12.3-common-rt/include/linux/sched.h:2146,
                 from /usr/src/linux-headers-6.12.3-common-rt/include/linux/percpu.h:12,
                 from /usr/src/linux-headers-6.12.3-common-rt/arch/x86/include/asm/msr.h:15,
                 from /usr/src/linux-headers-6.12.3-common-rt/arch/x86/include/asm/tsc.h:10,
                 from /usr/src/linux-headers-6.12.3-common-rt/arch/x86/include/asm/timex.h:6,
                 from /usr/src/linux-headers-6.12.3-common-rt/include/linux/timex.h:67,
                 from /usr/src/linux-headers-6.12.3-common-rt/include/linux/time32.h:13,
                 from /usr/src/linux-headers-6.12.3-common-rt/include/linux/time.h:60,
                 from /usr/src/linux-headers-6.12.3-common-rt/include/linux/stat.h:19,
                 from /usr/src/linux-headers-6.12.3-common-rt/include/linux/module.h:13,
                 from /var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.c:2:
/usr/src/linux-headers-6.12.3-common-rt/include/linux/spinlock_rt.h:25:9: note: macro "_spin_lock_init" defined here
   25 | #define _spin_lock_init(slock, percpu)                          \
      |         ^~~~~~~~~~~~~~~
/var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.c:181:52: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  181 | static inline void _spin_lock_init(_spinlock_t *s) {
      |                                                    ^
/var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.c: In function 'new_table':
/var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.c:932:40: error: macro "_spin_lock_init" requires 2 arguments, but only 1 given
  932 |         _spin_lock_init(&t->player_lock);
      |                                        ^
/usr/src/linux-headers-6.12.3-common-rt/include/linux/spinlock_rt.h:25:9: note: macro "_spin_lock_init" defined here
   25 | #define _spin_lock_init(slock, percpu)                          \
      |         ^~~~~~~~~~~~~~~
/var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.c:932:9: error: '_spin_lock_init' undeclared (first use in this function); did you mean '_spinlock_t'?
  932 |         _spin_lock_init(&t->player_lock);
      |         ^~~~~~~~~~~~~~~
      |         _spinlock_t
/var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.c:932:9: note: each undeclared identifier is reported only once for each function it appears in
/var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.c: In function 'launch_thread':
/var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.c:4205:39: error: macro "_spin_lock_init" requires 2 arguments, but only 1 given
 4205 |         _spin_lock_init(&tt->tree_lock);
      |                                       ^
/usr/src/linux-headers-6.12.3-common-rt/include/linux/spinlock_rt.h:25:9: note: macro "_spin_lock_init" defined here
   25 | #define _spin_lock_init(slock, percpu)                          \
      |         ^~~~~~~~~~~~~~~
/var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.c:4205:9: error: '_spin_lock_init' undeclared (first use in this function); did you mean '_spinlock_t'?
 4205 |         _spin_lock_init(&tt->tree_lock);
      |         ^~~~~~~~~~~~~~~
      |         _spinlock_t
/var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.c: In function 'play_stream':
/var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.c:4454:43: error: macro "_spin_lock_init" requires 2 arguments, but only 1 given
 4454 |         _spin_lock_init(&play_stream->lock);
      |                                           ^
/usr/src/linux-headers-6.12.3-common-rt/include/linux/spinlock_rt.h:25:9: note: macro "_spin_lock_init" defined here
   25 | #define _spin_lock_init(slock, percpu)                          \
      |         ^~~~~~~~~~~~~~~
/var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.c:4454:9: error: '_spin_lock_init' undeclared (first use in this function); did you mean '_spinlock_t'?
 4454 |         _spin_lock_init(&play_stream->lock);
      |         ^~~~~~~~~~~~~~~
      |         _spinlock_t
make[2]: *** [/usr/src/linux-headers-6.12.3-common-rt/scripts/Makefile.build:234: /var/lib/dkms/rtpengine/12.5.1.15/build/xt_RTPENGINE.o] Error 1
make[1]: *** [/usr/src/linux-headers-6.12.3-common-rt/Makefile:1961: /var/lib/dkms/rtpengine/12.5.1.15/build] Error 2
make: *** [/usr/src/linux-headers-6.12.3-common-rt/Makefile:236: __sub-make] Error 2
make: Leaving directory '/usr/src/linux-headers-6.12.3-rt-amd64'

```

**Comment by diederikdehaas** [comment]
This is probably caused by this upstream Linux commit:
[5c2e7736e20d](https://git.kernel.org/linus/5c2e7736e20d) ("rust: helpers: Avoid raw_spin_lock initialization for PREEMPT_RT")

and 'backported' to 6.12.2 in commit [00ccce23e159](https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.12.y&id=00ccce23e159ed50ece76bdfbb2d54f18fccc2b2)

**Comment by rfuchs** [comment]
Fixed by 6b3f3b969
